### DISCUSSION TITLE
Add support for OpenAPI 3.1 webhooks

### DISCUSSION THREAD

**Discussion by federicobond** [scalar/scalar#773] [Ideas]
Webhooks were introduced in OpenAPI 3.1. From the [spec](https://spec.openapis.org/oas/latest.html#fixed-fields):

> The incoming webhooks that MAY be received as part of this API and that the API consumer MAY choose to implement. Closely related to the callbacks feature, this section describes requests initiated other than by an API call, for [example](https://github.com/OAI/OpenAPI-Specification/tree/master/examples/v3.1/webhook-example.yaml) by an out of band registration. The key name is a unique string to refer to each webhook, while the (optionally referenced) Path Item Object describes a request that may be initiated by the API provider and the expected responses.

As an example, they show up like this in Redoc

<img width="299" alt="Captura de pantalla 2024-01-14 a la(s) 23 52 25" src="https://github.com/scalar/scalar/assets/138426/3aa2d272-f54f-442b-9d08-070bc6c1106f">

**Comment by marclave** [comment]
yes we definitely need to support this! for proper 3.1 support, thanks for opening up this discussion @federicobond 

@hanspagel @cameronrohani we will need to add another label for the request type, in this case "event". 

When we render webhooks, it should also be rendered above "Models" but below all the "Tags + Operations" :) 

Seems like we just need to add a "Webhooks" component with a few changes, but ultimately we can re-use most of the building blocks from the other components (body, try it out etc.) based on the official specification + example @federicobond provided.

the other part is we would need to add it to our offline search feature, but that shouldn't be too difficult ✨ 

to recap

- [ ] Add webhooks component that gets rendered above models
- [ ] Add webhooks rendering to sidebar
- [ ] Add new label for event (in sidebar)
- [ ] Add webhooks to Fuse.js search component

@federicobond @cameronrohani @hanspagel anything else im missing? :)

**Reply by federicobond** [reply]
Thanks for the quick response @marclave! One thing that I wanted to clarify from my example is that those webhooks are actually tagged "webhooks". They should probably appear alongside other operations in the more general case, at least that's what happens in Redoc.

**Comment by bazmandude** [comment]
Has there been any progress on adding webhook support?
